packages/compiler-core/src/transforms/vBind.ts

```diff
@@ -41,25 +42,11 @@ export const transformBind: DirectiveTransform = (dir, _node, context) => {
 
   // same-name shorthand - :arg is expanded to :arg="arg"
   if (!exp) {
-    if (arg.type !== NodeTypes.SIMPLE_EXPRESSION || !arg.isStatic) {
-      // only simple expression is allowed for same-name shorthand
-      context.onError(
-        createCompilerError(
-          ErrorCodes.X_V_BIND_INVALID_SAME_NAME_ARGUMENT,
-          arg.loc,
-        ),
-      )
-      return {
-        props: [
-          createObjectProperty(arg, createSimpleExpression('', true, loc)),
-        ],
-      }
-    }
+    if (!exp) {
```

Comment:
This seems repeated with L45

Reply:
Thank you for catching that oversight. I appreciate you taking the time to review the code and provide feedback. I missed removing that during the commit process, but I've now addressed it.